phpBB/phpbb/storage/file_info.php

```diff
@@ -62,14 +81,40 @@ public function __construct(adapter_interface $adapter, $path)
 	 */
 	public function get($name)
 	{
+		// Try to read properties from db if isn't set
 		if (!isset($this->properties[$name]))
 		{
-			if (!method_exists($this->adapter, 'file_' . $name))
+			$sql = 'SELECT metadata
+				FROM ' .  $this->storage_table . "
+				WHERE storage = '" . $this->db->sql_escape($this->storage_name) . "'
+					AND file_path = '" . $this->db->sql_escape($this->path) . "'";
+			$result = $this->db->sql_query($sql);
+			$metadata = json_decode($this->db->sql_fetchfield('metadata'), true);
+			$this->db->sql_freeresult($result);
+
+			if ($metadata !== null)
```

Comment:
That could be gone too, otherwise no metadata update if database say empty, or?!

Reply:
Uhm probably yes.

Reply:
I think maybe the code was better before

Reply:
This look is clear and without duplicates, I like it more.  👍